ui/pages/onboarding-flow/welcome/welcome.js

```diff
@@ -72,7 +79,62 @@ export default function OnboardingWelcome() {
     history,
     firstTimeFlowType,
     newAccountCreationInProgress,
+    dispatch,
   ]);
+
+  useEffect(() => {
+    const addNetworks = async () => {
+      if (!completedOnboarding) {
+        // List of chainIds to add (as hex strings)
+        const chainIdsToAdd = [
+          CHAIN_IDS.ARBITRUM,
+          CHAIN_IDS.BASE,
+          CHAIN_IDS.BSC,
+          CHAIN_IDS.OPTIMISM,
+          CHAIN_IDS.POLYGON,
+        ];
+
+        // Define the desired order based on `networkId`
+        const networkOrder = [
+          CHAIN_IDS.MAINNET,
+          CHAIN_IDS.ARBITRUM,
+          CHAIN_IDS.BASE,
+          CHAIN_IDS.BSC,
+          CHAIN_IDS.LINEA_MAINNET,
+          CHAIN_IDS.OPTIMISM,
+          CHAIN_IDS.POLYGON,
+        ];
+
+        // Filter the FEATURED_RPCS based on the chainIdsToAdd array
+        const selectedNetworks = FEATURED_RPCS.filter((network) =>
+          chainIdsToAdd.includes(network.chainId),
+        );
+
+        for (const network of selectedNetworks) {
+          await dispatch(
+            addNetwork({
+              chainId: network.chainId,
+              blockExplorerUrls: network.blockExplorerUrls,
+              defaultRpcEndpointIndex: network.defaultRpcEndpointIndex,
+              defaultBlockExplorerUrlIndex:
+                network.defaultBlockExplorerUrlIndex,
+              name: network.name,
+              nativeCurrency: network.nativeCurrency,
+              rpcEndpoints: network.rpcEndpoints,
+            }),
+          );
```

Comment:
Are these `Promises`? Wonder if `Promise.all` will help avoid some sort of race condition here

Reply:
totally agree with @gambinish . You can do like:
```
try {
  await Promise.all(
    selectedNetworks.map(async (network) => {
      const {
        chainId,
        blockExplorerUrls,
        defaultRpcEndpointIndex,
        defaultBlockExplorerUrlIndex,
        name,
        nativeCurrency,
        rpcEndpoints,
      } = network;

      await dispatch(
        addNetwork({
          chainId,
          blockExplorerUrls,
          defaultRpcEndpointIndex,
          defaultBlockExplorerUrlIndex,
          name,
          nativeCurrency,
          rpcEndpoints,
        }),
      );

      console.info(`Successfully added network: ${name}`);
    }),
  );
} catch (error) {
  console.error('Error adding networks:', error);
}

Reply:
fixed [31c62b9](https://github.com/MetaMask/metamask-extension/pull/28006/commits/31c62b9b107de2f90e14d20fadc040b78acd9528)